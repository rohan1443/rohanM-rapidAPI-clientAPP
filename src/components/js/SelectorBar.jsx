import React, { Component } from 'react'
import { fetchCountry, fetchLangLoc } from '../../actions'
import { connect } from 'react-redux'


export class SelectorBar extends Component {
  componentDidMount() {
    this.props.fetchCountry('US')
    this.props.fetchLangLoc('EN-US')
  }

  render() {
    const { country, langLoc } = this.props
    return (
      <div className="selection-bar-container">
        <div className="selection-bar-content">
          <div className="selector country"> {country} </div>
          <div className="selector langloc"> {langLoc} </div>
        </div>
      </div>
    )
  }
}

const mapStateToProps = ({ country, langLoc }) => {
  return { country, langLoc }
}

export default connect(mapStateToProps, { fetchCountry, fetchLangLoc })(SelectorBar)