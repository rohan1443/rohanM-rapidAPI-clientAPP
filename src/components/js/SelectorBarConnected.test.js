import React from 'react'
import renderer from 'react-test-renderer';
import configureMockStore from "redux-mock-store";
import thunk from 'redux-thunk'
import { Provider } from 'react-redux'
import sinon from 'sinon'
import { mount } from 'enzyme'

import SelectorBar from './SelectorBar'

const mockStore = configureMockStore([thunk])


describe("SelectorBar React component testing", () => {
  let store, wrapper, fetchCountry, fetchLangLoc, props

  beforeEach(() => {
    store = mockStore({})
    props =  {
      fetchCountry: () => {}
    , fetchLangLoc: () => {}
  }
  fetchCountry = sinon.stub(props, 'fetchCountry')
  fetchLangLoc = sinon.stub(props, 'fetchLangLoc')
  });

  afterEach(() => {
    fetchCountry.reset()
    fetchLangLoc.reset()
  })

  test("SelectorBar componenet rendered correctly without crashing", () => {
   wrapper = renderer.create(
      <Provider store={store}>
        <SelectorBar {...props} />
      </Provider>).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  test("testing reducer", () => {
    wrapper = mount(
      <Provider store={store}>
        <SelectorBar {...props} />
      </Provider>)

    const propsFromReduxCountryLocale = wrapper.find(SelectorBar).props()

    expect(propsFromReduxCountryLocale).toEqual({
        fetchCountry
      , fetchLangLoc
    })
  })
})

