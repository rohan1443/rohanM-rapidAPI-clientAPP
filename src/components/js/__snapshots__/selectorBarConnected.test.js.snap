// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectorBar React component testing SelectorBar componenet rendered correctly without crashing 1`] = `
<div
  className="selection-bar-container"
>
  <div
    className="selection-bar-content"
  >
    <div
      className="selector country"
    >
       
       
    </div>
    <div
      className="selector langloc"
    >
       
       
    </div>
  </div>
</div>
`;
